python/cuml/decomposition/pca.pyx

```diff
@@ -296,6 +299,11 @@ class PCA(Base):
         self._mean_ = None  # accessed via estimator.mean_
         self._noise_variance_ = None  # accessed via estimator.noise_variance_
 
+        # These variables control whether a sparse model was fit
+        # and what sparse type was used
+        self._sparse_model = False
```

Comment:
When testing this, I received an unexpected error that was not immediately obvious to me. This really shouldn't matter though, right? The components should be the same no matter if it was trained on sparse or dense. Why not just call the necessary function depending on the input that's passed in?

Reply:
As discussed offline, we will be waiting on this for better conversion between CumlArray and CuPy